### PR TITLE
chore: integrate unions

### PR DESCRIPTION
**Pull request by timsuchanek**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+	"devDependencies": {
+		"ts-node": "^9.0.0",
+		"typescript": "^4.0.2"
+	}
+}
```
Comment (Jolg42) [inline package.json:6]: Looks like you could delete this?

File: src/packages/client/src/__tests__/atomicOperationsUpdate.test.ts [hunk 2]
```diff
@@ -33,12 +38,8 @@ describe('minimal atomic update transformation', () => {
...
         updateOneUser(
           data: {
-            countFloat: {
-              set: 3.1415926
-            }
-            countInt1: {
-              set: 3
-            }
+            countFloat: 3.1415926
+            countInt1: 3
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/atomicOperationsUpdate.test.ts:42]: Set is not required anymore?

File: src/packages/client/src/__tests__/deepSome.test.ts [hunk 3]
```diff
@@ -40,18 +40,12 @@ describe('minimal where transformation', () => {
           posts: {
             some: {
               author: {
-                is: {
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/deepSome.test.ts:43]: is: is not required anymore?

File: src/packages/client/src/__tests__/atomicOperationsUpdate.test.ts (hunk 2 above)
Comment (timsuchanek) [reply src/packages/client/src/__tests__/atomicOperationsUpdate.test.ts:42]: Correct! Dom does that now already in the query engine. We got rid of all transformations.

File: src/packages/client/src/__tests__/deepSome.test.ts (hunk 3 above)
Comment (timsuchanek) [reply src/packages/client/src/__tests__/deepSome.test.ts:43]: Yes!

File: package.json (hunk 1 above)
Comment (timsuchanek) [reply package.json:6]: Indeed!

File: src/packages/client/src/__tests__/__snapshots__/aggregate.test.ts.snap [hunk 4]
```diff
@@ -48,11 +48,11 @@ Invalid \`prisma.user()\` invocation:
 Unknown arg \`mount\` in mount for type AggregateUser. Did you mean \`where\`? Available args:
 type aggregateUser {
   where?: UserWhereInput
-  orderBy?: UserOrderByInput
+  orderBy?: List<UserOrderByInput> | UserOrderByInput
   cursor?: UserWhereUniqueInput
   take?: Int
   skip?: Int
-  distinct?: UserDistinctFieldEnum
+  distinct?: List<UserDistinctFieldEnum>
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/__snapshots__/aggregate.test.ts.snap:55]: Look into this

File: src/packages/client/src/__tests__/__snapshots__/noArgs.test.ts.snap [hunk 5]
```diff
@@ -105,12 +113,12 @@ Invalid \`prisma.user()\` invocation:
 ?   AND?: UserWhereInput,
 ?   OR?: UserWhereInput,
 ?   NOT?: UserWhereInput,
-?   id?: Int,
-?   name?: String,
-?   email?: String,
+?   id?: IntFilter,
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/__snapshots__/noArgs.test.ts.snap:116]: Why not Int anymore

File: src/packages/client/src/__tests__/scalarWhereTransformation.test.ts [hunk 6]
```diff
@@ -365,11 +351,11 @@ describe('scalar where transformation', () => {
     }
   })
 
-  test('filter by enum', () => {
+  test.only('filter by enum', () => {
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/scalarWhereTransformation.test.ts:354]: Remove only

File: src/packages/client/src/__tests__/update.test.ts [hunk 7]
```diff
@@ -19,6 +19,7 @@ function getTransformedDocument(select) {
 describe('minimal update transformation', () => {
   beforeAll(async () => {
     dmmf = new DMMFClass(await getDMMF({ datamodel: blog }))
+    debugger
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/update.test.ts:22]: clean up

File: src/packages/client/src/runtime/error-types.ts [hunk 8]
```diff
@@ -86,7 +86,7 @@ export interface InvalidArgNameError {
 export interface MissingArgError {
   type: 'missingArg'
   missingName: string
-  missingType: DMMF.SchemaArgInputType[] // note that this could be an object or scalar type. in the object case, we print the whole object type
+  missingArg: DMMF.SchemaArg // note that this could be an object or scalar type. in the object case, we print the whole object type
```
Comment (timsuchanek) [inline src/packages/client/src/runtime/error-types.ts:89]: adjust comment

File: src/packages/client/src/runtime/query.ts [hunk 9]
```diff
@@ -769,8 +772,8 @@ export class Arg {
     this.error = error
     this.schemaArg = schemaArg
     this.isNullable =
-      schemaArg?.inputType.reduce<boolean>(
-        (isNullable, inputType) => isNullable && inputType.isNullable,
+      schemaArg?.inputTypes.reduce<boolean>(
+        (isNullable, inputType) => isNullable && schemaArg.isRequired,
```
Comment (timsuchanek) [inline src/packages/client/src/runtime/query.ts:776]: Why not isNullable?

File: src/packages/client/src/runtime/query.ts [hunk 10]
```diff
@@ -1555,32 +1329,27 @@ function valueToArg(key: string, value: any, arg: DMMF.SchemaArg): Arg | null {
...
     })
   }
 
-  if (value === null && arg.inputType.length === 1) {
-    const t = arg.inputType[0]
-    if (isInputArgType(t.type) && t.type.isOrderType) {
-      return null
-    }
-  }
+  // TODO: Maybe we have to enable this again
```
Comment (timsuchanek) [inline src/packages/client/src/runtime/query.ts:1339]: Can be removed

File: src/packages/client/src/runtime/query.ts [hunk 11]
```diff
@@ -1555,32 +1329,27 @@ function valueToArg(key: string, value: any, arg: DMMF.SchemaArg): Arg | null {
...
-    (arg.inputType.length > 1 ? arg.inputType[1].isNullable : false)
-  const isRequired =
-    arg.inputType[0].isRequired ||
-    (arg.inputType.length > 1 ? arg.inputType[1].isRequired : false)
   if (value === null && !isNullable && !isRequired) {
     // we don't need to execute this ternery if not necessary
-    const isAtLeastOne = isInputArgType(arg.inputType[0].type)
-      ? arg.inputType[0].type.atLeastOne
+    const isAtLeastOne = isInputArgType(arg.inputTypes[0].type)
+      ? (arg.inputTypes[0].type.constraints.minNumFields !== null && arg.inputTypes[0].type.constraints.minNumFields > 0)
```
Comment (timsuchanek) [inline src/packages/client/src/runtime/query.ts:1352]: Why just inputTypes[0]?

File: src/packages/client/src/runtime/query.ts [hunk 12]
```diff
@@ -1863,25 +1592,26 @@ function objectToArgs(
   }, [] as Arg[])
   // Also show optional neighbour args, if there is any arg missing
   if (
-    (entries.length === 0 && inputType.atLeastOne) ||
-    argsList.find((arg) => arg.error && arg.error.type === 'missingArg')
+    // (entries.length === 0 && inputType.constraints.minNumFields) ||
```
Comment (timsuchanek) [inline src/packages/client/src/runtime/query.ts:1595]: Clean up

File: src/packages/client/src/__tests__/__snapshots__/noArgs.test.ts.snap (hunk 5 above)
Comment (timsuchanek) [reply src/packages/client/src/__tests__/__snapshots__/noArgs.test.ts.snap:116]: It was a bug and is now fixed :)

File: src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap [hunk 13]
```diff
@@ -964,11 +964,11 @@ export type FindOnePostArgs = {
   /**
    * Select specific fields to fetch from the Post
   **/
-  select?: PostSelect | null
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap:967]: Check why `| null` is gone

File: src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap [hunk 14]
```diff
@@ -3428,14 +3428,14 @@ export type ManyRequiredGetPayload<
     ? ManyRequired  & {
       [P in TrueKeys<S['include']>]:
       P extends 'one'
-      ? OneOptionalGetPayload<S['include'][P]> | null : never
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap:3431]: Check why `| null` is gone

File: src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap [hunk 15]
```diff
@@ -4892,16 +4892,16 @@ export type OptionalSide2Args = {
 
 
 export type PostWhereInput = {
-  AND?: Enumerable<PostWhereInput>
```
Comment (timsuchanek) [inline src/packages/client/src/__tests__/integration/happy/exhaustive-schema/__snapshots__/test.ts.snap:4895]: This doesn't look right

**Pull request merged**